crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -572,6 +572,90 @@ impl Mesh {
+
+        // The indices of `other` should start after the last vertex of `self`.
+        let index_offset = self.count_vertices();
+
+        // Extend attributes of `self` with attributes of `other`.
+        for (id, values) in self.attributes_mut() {
+            let enum_variant_name = values.enum_variant_name();
+            if let Some(other_values) = other.attribute(id) {
+                match (values, other_values) {
+                    (Float32(vec1), Float32(vec2)) => vec1.extend(vec2),
+                    (Sint32(vec1), Sint32(vec2)) => vec1.extend(vec2),
+                    (Uint32(vec1), Uint32(vec2)) => vec1.extend(vec2),
+                    (Float32x2(vec1), Float32x2(vec2)) => vec1.extend(vec2),
+                    (Sint32x2(vec1), Sint32x2(vec2)) => vec1.extend(vec2),
+                    (Uint32x2(vec1), Uint32x2(vec2)) => vec1.extend(vec2),
+                    (Float32x3(vec1), Float32x3(vec2)) => vec1.extend(vec2),
+                    (Sint32x3(vec1), Sint32x3(vec2)) => vec1.extend(vec2),
+                    (Uint32x3(vec1), Uint32x3(vec2)) => vec1.extend(vec2),
+                    (Sint32x4(vec1), Sint32x4(vec2)) => vec1.extend(vec2),
+                    (Uint32x4(vec1), Uint32x4(vec2)) => vec1.extend(vec2),
+                    (Float32x4(vec1), Float32x4(vec2)) => vec1.extend(vec2),
+                    (Sint16x2(vec1), Sint16x2(vec2)) | (Snorm16x2(vec1), Snorm16x2(vec2)) => {
+                        vec1.extend(vec2);
+                    }
+                    (Uint16x2(vec1), Uint16x2(vec2)) | (Unorm16x2(vec1), Unorm16x2(vec2)) => {
+                        vec1.extend(vec2);
+                    }
+                    (Sint16x4(vec1), Sint16x4(vec2)) | (Snorm16x4(vec1), Snorm16x4(vec2)) => {
+                        vec1.extend(vec2);
+                    }
+                    (Uint16x4(vec1), Uint16x4(vec2)) | (Unorm16x4(vec1), Unorm16x4(vec2)) => {
+                        vec1.extend(vec2);
+                    }
+                    (Sint8x2(vec1), Sint8x2(vec2)) | (Snorm8x2(vec1), Snorm8x2(vec2)) => {
+                        vec1.extend(vec2);
+                    }
+                    (Uint8x2(vec1), Uint8x2(vec2)) => {
+                        vec1.extend(vec2);
+                    }
+                    (Unorm8x2(vec1), Unorm8x2(vec2)) => vec1.extend(vec2),
+                    (Sint8x4(vec1), Sint8x4(vec2)) => {
+                        vec1.extend(vec2);
+                    }
+                    (Snorm8x4(vec1), Snorm8x4(vec2)) => vec1.extend(vec2),
+                    (Uint8x4(vec1), Uint8x4(vec2)) => {
+                        vec1.extend(vec2);
+                    }
+                    (Unorm8x4(vec1), Unorm8x4(vec2)) => vec1.extend(vec2),
+                    _ => panic!(
+                        "Incompatible vertex attribute types {} and {}",
+                        enum_variant_name,
+                        other_values.enum_variant_name()
+                    ),
```

Comment:
We might want to return some error instead? I'm not sure how common it would be to reach this panic since I'd expec e.g. position attributes to almost always be `Float32x3`